src/eosjs2-api.ts

```diff
@@ -93,26 +93,49 @@ export class Api {
         }));
     }
 
-    async pushTransaction({ blocksBehind, expireSeconds, actions, ...transaction }: any) {
-        let info: GetInfoResult;
-        if (!this.chainId) {
-            info = await this.rpc.get_info();
-            this.chainId = info.chain_id;
-        }
-        if (blocksBehind !== undefined && expireSeconds !== undefined) {
-            if (!info)
-                info = await this.rpc.get_info();
-            let refBlock = await this.rpc.get_block(info.head_block_num - blocksBehind);
-            transaction = { ...ser.transactionHeader(refBlock, expireSeconds), ...transaction };
+    // eventually break out into TransactionValidator class
+    private hasRequiredTaposFields({ expiration, ref_block_num, ref_block_prefix, ...transaction}: any): boolean {
+       return !!(expiration && ref_block_num && ref_block_prefix);
+    }
+
+    async transact(transaction: any, config: TransactionConfig): Promise<any>  {
+      let info: GetInfoResult;
+      const { broadcast, blocksBehind, expireSeconds } = { broadcast: true, ...config };
```

Comment:
I like the syntax, but this line may warrant a comment: 

`// sets broadcast to true by default; overwrites that value if provided in config`

...or something like that.

Reply:
I don't care for comments which document language.